src/routes/api/auth/self/+server.ts

```diff
@@ -68,29 +68,30 @@ export const PUT: RequestHandler = async ({ request, locals }) => {
 	// upload avatar (get storage URL)
 	let avatar_id;
 	try {
-		avatar_id = await upload_avatar(
-			locals.crystal.username,
-			Buffer.from(payload.avatar, "base64"),
-		);
+		if (payload.avatar.length > 8 * 1024 * 1024) {
+			throw new Error();
+		}
+		if (payload.avatar.length > 64) {
+			avatar_id = await upload_avatar(
+				locals.crystal.username,
+				Buffer.from(payload.avatar, "base64"),
+			);
+		}
```

Comment:
If the avatar is not uploaded, maybe we should throw an error? so that it's more clear what happened.

Reply:
Yes, that's a good idea, but I would like to allow empty (or optional) field to reduce the payload size if we are not going to update the avatar.